### PR TITLE
communities: added funding to community settings.

### PR DESCRIPTION
**Pull request by alejandromumo**
- requires https://github.com/inveniosoftware/invenio-communities/pull/531 Otherwise the `funding` relation is not going to be denormalised and the UI won't render funders/awards properly.
- requires https://github.com/inveniosoftware/react-invenio-deposit/pull/465 being released. Otherwise the component will not be found. 
- closes https://github.com/inveniosoftware/invenio-communities/issues/587

this PR adds `FundingField` to community settings.

### CODE REVIEW

File: invenio_communities/webpack.py [hunk 1]
```diff
@@ -54,6 +54,8 @@
                 '@ckeditor/ckeditor5-react': '^2.1.0',
                 '@semantic-ui-react/css-patch': '^1.0.0',
                 'react-invenio-forms': '^0.10.0',
+                # TODO update after releasing Funding Field
+                'react-invenio-deposit': '^0.19.4',
```
Comment (alejandromumo) [inline invenio_communities/webpack.py:58]: This should be setup to the correct version after https://github.com/inveniosoftware/react-invenio-deposit/pull/465/files gets released

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 2]
```diff
@@ -256,18 +258,124 @@ class CommunityProfileForm extends Component {
...
     });
 
+    /**
+     * Deserializes a funding record (e.g. funder or award)
+     *
+     * @param {object} fund
+     *
+     * @returns {object} an object containing the deserialized record
+     */
+     const deserializeFunding = ((fund) => {
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:268]: This is exactly the same logic to deserialise a funding record that was used in  https://github.com/inveniosoftware/react-invenio-deposit/blob/aefebaa2156c752ee10829d69c85b3ab22a15eeb/src/lib/fields/VocabularyField.js#L42-L44

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 3]
```diff
@@ -256,18 +258,124 @@ class CommunityProfileForm extends Component {
...
   serializeValues = (values) => {
+    /**
+     * Serializes a funding record (e.g. funder or award)
+     *
+     * @param {object} fund
+     *
+     * @returns {object} an object containing the serialized record
+     */
+     const serializeFunding = ((fund) => {
+      const _serialize = (value) => {
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:343]: This is exactly the same logic to serialise a funding record that was used in

https://github.com/inveniosoftware/react-invenio-deposit/blob/aefebaa2156c752ee10829d69c85b3ab22a15eeb/src/lib/fields/VocabularyField.js#L296

Comment (zzacharo) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:343]: Can you just import the field and use the serialize method? Maybe it's not exported but we could export the field eventually....

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 4]
```diff
@@ -426,27 +541,106 @@ class CommunityProfileForm extends Component {
...
                     allowAdditions
                     search={(filteredOptions, searchQuery) => filteredOptions}
                   />
+                  <FundingField
+                    fieldPath="metadata.funding"
+                    searchConfig={{
+                      searchApi: {
+                        axios: {
+                          headers: {
+                            //  FIXME use for internationalisation
```
Comment (zzacharo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:559]: Shall we fix that? Or create an issue and delete the comment?

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 5]
```diff
@@ -432,6 +550,71 @@ class CommunityProfileForm extends Component {
...
+                    }
+                    computeFundingContents={(funding) => {
+                      let headerContent, descriptionContent = '';
+                      let awardOrFunder = 'award';
+                      if (funding.award) {
+                        headerContent = funding.award.title;
+                      }
+
+                      if (funding.funder) {
+                        const funderName = funding.funder?.name ?? funding.funder?.title?.en ?? funding.funder?.id ?? '';
```
Comment (zzacharo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:606]: Shouldn't the i18n title come first? Is the funder a person or organization type? If so, isn't it better to check explicitly what type of funder you have and be stricter with where to check for the name?

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js (hunk 2 above)
Comment (zzacharo) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:268]: Can you just import the field and use the deserialize method? Maybe it's not exported but we could export the field eventually....

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 6]
```diff
@@ -432,6 +550,71 @@ class CommunityProfileForm extends Component {
...
+                    fieldPath="metadata.funding"
+                    searchConfig={{
+                      searchApi: {
+                        axios: {
+                          headers: {
+                            //  FIXME use for internationalisation
+                            //  Accept: "application/vnd.inveniordm.v1+json"
+                            Accept: "application/json",
+                          },
+                          url: "https://127.0.0.1:5000/api/awards",
```
Comment (zzacharo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:563]: 👀

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js (hunk 4 above)
Comment (alejandromumo) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:577]: I think the backend for funders and grants is not ready yet to internationalise (e.g. `funder.title` should be declared as a `L10NString`).

**Pull request merged**